lib/jekyll/tags/highlight.rb

```diff
@@ -39,6 +40,17 @@ def initialize(tag_name, markup, tokens)
         end
       end
 
+      def parse(tokens)
+        if @lang == 'raw'
+          end_pos = tokens.rindex{|x| x =~ END_HIGHLIGHT }
```

Comment:
This isn't going to work in liquid 4, since it passes a Liquid::Tokenizer instead of an array as the argument to `parse`.  Does jekyll do something like this elsewhere?  Normally all that is needed is to call `tokens.shift`